### PR TITLE
feat!: Drop php7.4 and support php 8.2 and 8.3

### PR DESCRIPTION
**Pull request by fibble**
**Breaking Change**

| Q                | A
| ---------------- | ---
| Type             | Feature
| Ticket           | VOL-4761

**Description of the change:**
Drops 7.4 support and some changes for 8.2 and 8.3 support.

### CODE REVIEW

File: src/Exception/ApiException.php [hunk 1]
```diff
@@ -6,15 +6,12 @@
...
     public function __construct(
         string    $message = '',
         int       $code = 0,
         Throwable $previous = null,
-        array     $data = []
+        public array     $data = []
     ) {
-        $this->data = $data;
         parent::__construct($message, $code, $previous);
     }
```
Comment (hobbyhacker0) [inline src/Exception/ApiException.php:16]: So `$data` wasn't set in the construct before, and still isn't?
Can't it be removed?

**Review by ilindsay — APPROVED** [review]

File: src/Exception/ApiException.php (hunk 1 above)
Comment (fibble) [reply src/Exception/ApiException.php:16]: It was previously being set in the constructor. Moving to constructor property promotion pattern.

File: src/Provider/GovUkAccount.php [hunk 2]
```diff
@@ -68,7 +65,7 @@ public function __construct(
     /**
      * Parses a JWK in array format, returning a Key object or throwing an exception.
      *
-     * @param array|string $jwk JWK - Will attempt to parse string as JSON.
```
Comment (JoshuaLicense) [inline src/Provider/GovUkAccount.php]: This method supports a string parameter.

File: src/Provider/GovUkAccount.php [hunk 3]
```diff
@@ -451,7 +449,11 @@ protected function createResourceOwner(
         if (!$encoded) {
             throw new \JsonException('Could not encode $response');
         }
-        $response = json_decode($encoded, true);
+
+        $response = json_decode($encoded, true, 512);
```
Comment (JoshuaLicense) [inline src/Provider/GovUkAccount.php]: Do we need to define depth?

```suggestion
        $response = json_decode($encoded, true);
```

File: src/Provider/GovUkAccountUser.php [hunk 4]
```diff
@@ -24,16 +21,14 @@ public function getId()
...
-     *
      * @return mixed|null
      */
     public function getField(string $key)
     {
         return $this->data[$key] ?? null;
     }
 
-    public function jsonSerialize()
+    public function jsonSerialize(): mixed
```
Comment (JoshuaLicense) [inline src/Provider/GovUkAccountUser.php]: Could this be typed as an `array`?

File: test/Provider/GovUkAccountTest.php [hunk 5]
```diff
@@ -440,7 +440,7 @@ protected function createAccessToken(array $payload = []): string
             'client_id' => 'mock_client_id',
             'exp' => (new \DateTimeImmutable())->getTimestamp() + 3600,
             'iat' => (new \DateTimeImmutable())->getTimestamp() - 10,
-        ], $payload), base64_decode(static::SERVICE_PRIVATE_KEY), 'ES256', 'N-NegtAr3bnZC4d2Pcav2sH5UNOp5tpbXg8phQl4Tyg');
+        ], $payload), base64_decode((string) static::SERVICE_PRIVATE_KEY), 'ES256', 'N-NegtAr3bnZC4d2Pcav2sH5UNOp5tpbXg8phQl4Tyg');
```
Comment (JoshuaLicense) [inline test/Provider/GovUkAccountTest.php]: Aren't these already strings?

**Review by JoshuaLicense — CHANGES_REQUESTED** [review]
Looks good, a few minor things!

**Review by hobbyhacker0 — APPROVED** [review]

File: src/Provider/GovUkAccount.php [hunk 6]
```diff
@@ -15,6 +15,7 @@
 use GuzzleHttp\Psr7\HttpFactory;
 use Illuminate\Support\Collection;
 use InvalidArgumentException;
+use LanguageServerProtocol\PackageDescriptor;
```
Comment (JoshuaLicense) [inline src/Provider/GovUkAccount.php]: Is this used?

Comment (fibble) [reply src/Provider/GovUkAccount.php]: removed!

**Review by JoshuaLicense — APPROVED** [review]

**Pull request merged**